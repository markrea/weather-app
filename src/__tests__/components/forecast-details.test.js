import React from "react";
import { cleanup, render, } from "@testing-library/react";
import ForecastDetails from "../../components/forecast-details";

afterEach(cleanup);

describe("ForecastDetails", () => {
    const forecasts =
    {
        date: 1525046400000,
        temperature: {
            max: 11,
            min: 4,
        },
        wind: {
            speed: 30,
            direction: "s"
        },
        humidity: 30,

    };

    it("renders correctly to match ForecastDetails snapshot", () => {
        const { asFragment } = render(<ForecastDetails forecasts={forecasts} />);
        expect(asFragment).toMatchSnapshot();
    });


    it("renders the correct amount of ForecastSummary component props", () => {
        const { getByTestId } = render(
            <ForecastDetails forecasts={forecasts} />);

        expect(getByTestId("date-id")).toHaveClass("date");
        expect(getByTestId("minTemperature-id")).toHaveClass("minTemperature");
        expect(getByTestId("maxTemperature-id")).toHaveClass("maxTemperature");
        expect(getByTestId("humidity-id")).toHaveClass("humidity");
        expect(getByTestId("windSpeed-id")).toHaveClass("windSpeed");
        expect(getByTestId("windDirection-id")).toHaveClass("windDirection");

    });
});