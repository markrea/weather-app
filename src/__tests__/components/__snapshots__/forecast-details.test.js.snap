// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ForecastDetails renders correctly to match ForecastDetails snapshot 1`] = `[Function]`;
